### PR TITLE
fix reproducible noise bug

### PR DESCRIPTION
**Pull request by GarethDaviesGW**
Timeseries with different epochs (+ 10s difference) before bugfix:
image

Timeseries with different epochs after bugfix:
image

(Bug found by me, fix as suggested by @spxiwh)

### CODE REVIEW

File: pycbc/noise/reproduceable.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def normal(start, end, sample_rate=16384, seed=0):
     sv = RandomState(seed).randint(-2**50, 2**50)
     data = numpy.concatenate([block(i + sv, sample_rate)
                               for i in numpy.arange(s, e + 1, 1)])
```
Comment (spxiwh) [inline pycbc/noise/reproduceable.py:83]: Does this need to be `e+2`?

**Review by ahnitz — APPROVED** [review]
After considering the change. I agree. It should have the epoch set to the start of the *block* not the requested start time. The slicing then gets the requested range after that.

File: pycbc/noise/reproduceable.py (hunk 1 above)
Comment (ahnitz) [reply pycbc/noise/reproduceable.py:83]: @spxiwh Why? e + 1 should have an end time that is beyond the requested end. It's only the block at 'e' that might end earlier due to rounding. If this were a problem, we would have seen errors before when generated lots of fake noise.

**Review by ahnitz — CHANGES_REQUESTED** [review]
@GarethDaviesGW However, I do think I found one thing that does need to change. The 's' / 'e' blocks need to use the 'floor' operation and not 'int'.The edge case is if a block need to intersect time 0. (which may occur in tests...)

**Comment by ahnitz** [comment]
@GarethDaviesGW Would you be OK with adding a unittest test to test_noise to cover this issue?

File: pycbc/noise/reproduceable.py (hunk 1 above)
Comment (spxiwh) [reply pycbc/noise/reproduceable.py:83]: Hmm, yeah, think I miscounted. `e+1` does seem right.

**Comment by GarethDaviesGW** [comment]
> @GarethDaviesGW Would you be OK with adding a unittest test to test_noise to cover this issue?

I should be able to add that

File: pycbc/noise/reproduceable.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def normal(start, end, sample_rate=16384, seed=0):
     # This is reproduceable because we used fixed seeds from known values
     block_dur = BLOCK_SAMPLES / sample_rate
     s = int(start / block_dur)
```
Comment (ahnitz) [inline pycbc/noise/reproduceable.py:74]: @GarethDaviesGW Should not floor also be here? In fact, I think that's the immediate source of the error in the current test suite.

File: test/test_noise.py [hunk 3]
```diff
@@ -65,6 +65,11 @@ def test_noise_psd(self):
         ave = ratio.numpy().mean()
         self.assertAlmostEqual(ave, 1, 1)
 
+    def test_noise_reproducible(self):
+        ts1 = normal(20, 30, sample_rate=16384, seed=87693)
+        ts2 = normal(25, 35, sample_rate=16384, seed=87693)
+        self.assertEqual(ts1.time_slice(25, 30), ts2.time_slice(25, 30))
```
Comment (ahnitz) [inline test/test_noise.py:71]: @GarethDaviesGW Does assert equal here really know how to compare time series? (also there may be slight numerical error, so an 'almost equal' may be more appropriate. You may need to take the diff and check that the maximum difference is below some reasonable numerical threshold.

Comment (GarethDaviesGW) [reply test/test_noise.py:71]: assertequal was working here, I had used ts1(...) == t2(...), but the "False is not True" error output wasn't helpful

File: pycbc/noise/reproduceable.py (hunk 2 above)
Comment (GarethDaviesGW) [reply pycbc/noise/reproduceable.py]: yup - I realised that after seeing the error myself

File: test/test_noise.py (hunk 3 above)
Comment (GarethDaviesGW) [reply test/test_noise.py:71]: if it fails, the assertequal gives:
```
AssertionError: <pycbc.types.timeseries.TimeSeries object at 0x7fc563788790> != <pycbc.types.timeseries.TimeSeries object at 0x7fc5637880d0>
```

Which is (slightly) more descriptive!

**Comment by ahnitz** [comment]
@GarethDaviesGW The remaining failure, you'll need to update the presaved result for. Given the time series is changed, I believe this is expected that you'd need to do so.

**Review by ahnitz — APPROVED** [review]

**Comment by GarethDaviesGW** [comment]
Currently making a python3 venv to change the expected value there, so should be ready after finding out what that should be!

File: examples/live/check_results.py [hunk 4]
```diff
@@ -173,7 +173,7 @@ def check_coinc_results(args):
                     and close(inj_mass2[i], trig_props['mass2'][j], 5e-7)
                     and close(inj_spin1z[i], trig_props['spin1z'][j], 5e-7)
                     and close(inj_spin2z[i], trig_props['spin2z'][j], 5e-7)
-                    and close(15.0, trig_props['net_snr'][j], 1.0)):
+                    and close(13.0, trig_props['net_snr'][j], 1.0)):
```
Comment (titodalcanton) [inline examples/live/check_results.py:176]: @GarethDaviesGW note that there are two injections, and both need to pass this test (one improvement to the test would be to actually calculate the optimal SNRs on its own for each injection, but that is not in place yet).

Comment (GarethDaviesGW) [reply examples/live/check_results.py]: hadnt realised that - Ive updated to use the window of 2 in SNR instead
